File: v-kotlin/v-mongo/src/main/kotlin/model/Foobar.kt
package model

import org.bson.codecs.pojo.annotations.BsonId
import org.bson.types.ObjectId

data class Foobar (
  @BsonId
  val id: ObjectId? = null,
  val foo: Foo,
  val barList: List<Bar>
)

data class Foo (
  val fooId: String
)

data class Bar (
  val barId: String
)

data class FoobarUnwind (
  val foo: Foo,
  val bar: Bar
)

File: v-lang/l-go/built-in/strcutx/foobar.go
package strcutx

import "fmt"

type Foo struct {
	items []string
}

func (r Foo) Add() {
	r.items = append(r.items, "B")
	r.items = append(r.items, "C")
}

func (r Foo) Print() {
	fmt.Println(r.items)
}

// ================================================

type Bar struct {
	items []string
}

func (r *Bar) Add() {
	r.items = append(r.items, "B")
	r.items = append(r.items, "C")
}

func (r *Bar) Print() {
	fmt.Println(r.items)
}
